extensions/resteasy-classic/resteasy-jackson/deployment/src/main/java/io/quarkus/resteasy/jackson/deployment/ResteasyJacksonProcessor.java

```diff
@@ -4,11 +4,17 @@
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 
 public class ResteasyJacksonProcessor {
 
     @BuildStep
     void feature(BuildProducer<FeatureBuildItem> feature) {
         feature.produce(new FeatureBuildItem(Feature.RESTEASY_JACKSON));
     }
+
+    @BuildStep
+    ReflectiveClassBuildItem register() {
+        return new ReflectiveClassBuildItem(true, false, "com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector");
```

Comment:
IMHO Jackson stuff should be managed by the Jackson extension, which makes more sense here as  `resteasy-jackson` depends on  `jackson`.

Reply:
In theory I agree, but the thing is that the `jackson` extension does not depend on `com.fasterxml.jackson.module:jackson-module-jaxb-annotations` so there seems no need for it to register classes in it. On the other hand, `resteasy-jackson` does depend on `com.fasterxml.jackson.module:jackson-module-jaxb-annotations` (and is the reason we need to register the classes for reflection) so in practice it makes sense to me to move the registration there **unless** we expect people to manually add `com.fasterxml.jackson.module:jackson-module-jaxb-annotations` as a depedency to their project and assume that using `quarkus-jackson` would suffice to support it.

The Quarkus team should have better insight on this than me, since they probably have a better picture of how these extensions are used in the wild.

The alternative here is to keep the registration in `quarkus-jackson` and only perform it when `com.fasterxml.jackson.module:jackson-module-jaxb-annotations` is in the classpath.